ls1intum/Artemis
# [Enhancement] Change submit button to also save serverside

### Checklist
- [x] Client: I updated multiple integration tests (Jest) related to the features

### Motivation and Context



When clicking submit in the online editor with unsaved changes, a save request is issues before the commit request. This may lead to wrong behaviour in some edge cases with bad / no internet connection.

### Description


The 'save' and 'submit' actions are now handled in one request. The saveChangedFiles request does now optionally also 
commit the changes, depending on a query parameter.

### Steps for Testing


1. Log in to Artemis
2. Navigate to a programming exercise
3. Commit with and without unsaved changes
4. Check that everything works correctly
5. Inspect the network requests to be only one for saving and committing.

**Test also in the instructor editor with all repositories (Test, Solution, Template, ...)**
